### PR TITLE
adds CmdLine() exception handler

### PR DESCRIPTION
**Pull request by jhseol**
If /proc/pid/cmdline has no content, slice operation of the 'data' will be panicked. (e.g. kernel threads)

### CODE REVIEW

**Comment by beorn7** [comment]
:+1: but let's have @grobie take a look to verify.

File: proc.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ func (p Proc) CmdLine() ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	
+	if len(data) < 1 {
+		return nil, fmt.Errorf("could not read cmdline pid:%d", p.PID)
```
Comment (juliusv) [inline proc.go]: Could you do: `"could not read cmdline for pid: %d"`

Comment (juliusv) [inline proc.go]: Actually, should this really return an error, or just an empty command-line?

Comment (jhseol) [inline proc.go]: Hmm, just return an empty command-line make more sense to me.

Comment (juliusv) [inline proc.go]: Yeah. Curious - are you using this function from somewhere? Seems like the "prometheus" github org is not even using it anywhere yet.

Comment (grobie) [inline proc.go]: :+1: for an empty string

**Comment by grobie** [comment]
Please add a test case as well.

File: proc.go (hunk 1 above)
Comment (jhseol) [inline proc.go]: Okay, I'm building some kind of system event listener including docker and kernel proc events. 
As a part of the listener, I found this repository for proc fs reader. 
And we're planning to use prometheus for our system metric server. :)

**Comment by jhseol** [comment]
I added a test case for an empty cmdline file.

File: proc.go [hunk 2]
```diff
@@ -95,6 +95,10 @@ func (p Proc) CmdLine() ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+	
+	if len(data) < 1 {
+		return nil, nil
```
Comment (grobie) [inline proc.go]: I think we should return `[]string{}, nil` here, as the assumption in go for `type, error` return signatures is usually: if `err` is `nil` then the returned type is an initialized value.

File: proc_test.go [hunk 3]
```diff
@@ -44,12 +44,25 @@ func TestCmdLine(t *testing.T) {
...
+
+	p2, err := testProcess(26232)
+	if err != nil {
+		t.Fatal(err)
+	}
+	c2, err := p2.CmdLine()
+	if err != nil {
+		t.Fatal(err)
+	}
+	var want []string
```
Comment (grobie) [inline proc_test.go]: What I "want" here is actually `[]string{}`.

**Comment by jhseol** [comment]
Thanks for your comment.

**Comment by grobie** [comment]
:+1: Great, thanks! Now comes the last pedantry thingy ;), we usually tend to squash our commits together at the end of a PR review, so that we have only a few coherent commits with all changes to code+tests+fixtures together. I'll merge immediately after squashing them to a nice commit.

**Comment by jhseol** [comment]
:+1: Great!

**Comment by juliusv** [comment]
@jhseol Hey hey, but the commits are still not squashed - could you squash them? Then we'll merge :)

**Comment by jhseol** [comment]
Commits are squashed. Is it okay to merge?

**Comment by juliusv** [comment]
@jhseol Hmm, somehow the "Update example" commit by @audebert made it into your squashed commits (https://github.com/prometheus/procfs/pull/5/commits), but under a different SHA than in master. This should fix it:

```
git checkout master
git reset --hard origin/master
git cherry-pick 99d4f8f  # The one commit we're actually interested in.
git push -f origin master
```

**Comment by jhseol** [comment]
Commits are squashed again. I'm not sure why "Update example" commit was squashed with my commit. :(

**Comment by juliusv** [comment]
@jhseol no worries - looking good now :) Thanks! Merging.

**Pull request merged**